File: NoteMakingApplication/src/main/java/com/DXCTrainingProject/NoteMakingApplication/Service/AdminService.java
package com.DXCTrainingProject.NoteMakingApplication.Service;

import com.DXCTrainingProject.NoteMakingApplication.DTO.AdminNoteResponseDTO;
import com.DXCTrainingProject.NoteMakingApplication.DTO.UserResponseDTO;
import com.DXCTrainingProject.NoteMakingApplication.Entity.User;

import java.util.List;

public interface AdminService {

    List<UserResponseDTO> getAllUsers();
    List<AdminNoteResponseDTO> getAllNotes();
    String deleteUser(Long id);
    String deleteNote(Long id);
}

File: note-making-frontend/src/app/services/admin.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { AuthService } from './auth.service';
import { Observable } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class AdminService {
  private apiUrl = 'http://localhost:8080/api/admin';

  constructor(private http: HttpClient, private authService: AuthService) {}

  private getHeaders() {
    const token = this.authService.getToken();
    return {
      headers: new HttpHeaders({
        Authorization: `Bearer ${token}`,
      }),
    };
  }

  getAllUsers(): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/users`, this.getHeaders());
  }


  getAllNotes(): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/notes`, this.getHeaders());
  }


  deleteUser(id: number): Observable<any> {
    return this.http.delete(`${this.apiUrl}/users/${id}`, {
        ...this.getHeaders(),
        responseType: 'text' as 'json'
    });
  }

  deleteNote(id: number): Observable<any> {
    return this.http.delete(`${this.apiUrl}/notes/${id}`, {
        ...this.getHeaders(),
        responseType: 'text' as 'json'
    });
  }
}
